docs/db-migration-guide.ipynb

```diff
@@ -0,0 +1,137 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Charmed Kubeflow Databases Migration Guide\n",
+    "\n",
+    "There are two databases in Charmed Kubeflow - `katib-db` and `kfp-db`. In Charmed Kubeflow 1.7 these MariaDB databases deployed by `charmed-osm-mariadb-k8s` charm. Since release 1.7 there is an option to relate charm to new MySQL database that is deployed by `mysql-k8s` charm. In cases where previous deployment has MariaDB a migration of database is required."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "In order to perform migaraion the following tools should be installed on client machine that has access to Charmed Kubeflow deployment:\n",
+    "- `mysql-client` (install by running `sudo apt install mysql-client`)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Katib DB migration"
```

Comment:
It looks like the Katib DB migration is exactly the same as the KFP DB migration, I think we could avoid duplication by just adding a generic set of steps and tell people that they work for both, just the name of the application and unit is going to be different.

Reply:
I'm not sure.

This actually raises a difficulty that we never thought about before in terms of testable docs. If our doc testing process behind the scenes simply runs all the code blocks against a CKF deployment, then it will force us to write docs in this way - with exhaustive concrete sequences of commands that the user must run. But this conflicts with the readability of the docs, which would read better as Daniela suggested.

So we either have to:

1. Sacrifice readability for the sake of testability
2. Sacrifice testability for the sake of readability
3. Implement a more complex doc testing process e.g. we could present abstract steps to the user

Reply:
Not really identical. We get credentials differently using different unit names and different commands. Also, I could not get Katib DB populated properly though, so there could be some additional work to be done when we have populated Katib DB, like porting constraints, etc.


RE: testability.
We can have sections of doc executed multiple times with different parameters. Eg. we can put something in comments to indicate to test script that different DB name can be used, for example. This is whole new Epic though.

Reply:
Cool. My vote is leave it as it is.

Reply:
@DnPlas I pushed the change. I think, it made it more confusing when we try to create generilize. Please review and let me know if this works. I am not convinced that this is better than outlining each DB migration for each DB.

Reply:
Alright, it's okay with me